### PR TITLE
[cli] Move bin.ts core into -cli/run.ts

### PR DESCRIPTION
**Pull request by paulirish**
(partial) fix for #2601 

Unblocks issues like https://github.com/addyosmani/webpack-lighthouse-plugin/issues/5

No functional changes was made in this PR; it was only a copypaste party.

### CODE REVIEW

**Comment by patrickhulce** [comment]
needs a `yarn format`?

**Comment by paulirish** [comment]
@brendankenny test added: https://github.com/GoogleChrome/lighthouse/pull/2602/files#diff-4ebc963ff5871b2da4c3ed52bb31ad31

**Comment by brendankenny** [comment]
need a rebase (not sure whats going on with codecov)

File: lighthouse-cli/run.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+    return await launchedChrome.kill();
+  } catch (err) {
+    if (typeof launchedChrome !== 'undefined') {
+      await launchedChrome!.kill();
+    }
+
+    return handleError(err);
+  }
+}
```
Comment (brendankenny) [inline lighthouse-cli/run.ts]: new line

File: lighthouse-cli/test/cli/run-test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/**
+ * @license Copyright 2016 Google Inc. All Rights Reserved.
```
Comment (brendankenny) [inline lighthouse-cli/test/cli/run-test.js]: 2017

File: lighthouse-cli/cli-flags.ts [hunk 3]
```diff
@@ -17,8 +17,10 @@ export interface Flags {
       view: boolean, maxWaitForLoad: number
 }
 
-export function getFlags() {
-  return yargs.help('help')
+export function getFlags(fakeInput?: string) {
```
Comment (brendankenny) [inline lighthouse-cli/cli-flags.ts]: nice, but can we call the parameter something else? It seems like a legitimate use, not just for testing, so what about `manualInput`, `manualArgv`, `argvString`, or something something :)

**Comment by paulirish** [comment]
@brendankenny sg. done. 
rebased.

File: lighthouse-cli/test/cli/run-test.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  'settings': {
+    'onlyAudits': ['viewport']
+  }
+};
+
+const getFlags = require('../../cli-flags').getFlags;
+
+describe('CLI run', function() {
+  it('runLighthouse completes a LH round trip', () => {
+    const url = 'chrome://version';
```
Comment (brendankenny) [inline lighthouse-cli/test/cli/run-test.js]: :)

**Pull request merged**